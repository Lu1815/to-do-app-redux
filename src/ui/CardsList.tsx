import Card from './Card'
import { ITodoState } from '../App/redux/Index'

interface ICardsListProps {
    todosList: ITodoState[],
}

const CardsList = ({todosList}: ICardsListProps) => {
  return (
    <div className='flex flex-col justify-center gap-2 py-3'>
        {
          todosList.length === 0 ? <h1>There are no tasks yet...</h1>
          :
            todosList.map((todoItem) => (
                <Card text={todoItem.name} key={todoItem.id}/>
            ))
        }
    </div>
  )
}

export default CardsList