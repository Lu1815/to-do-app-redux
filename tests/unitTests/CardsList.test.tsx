import React from 'react';
import { render, screen } from '@testing-library/react';
import CardsList from '../../src/ui/CardsList';

// Mock the Card component since we want to test CardsList in isolation
jest.mock('../../src/ui/Card', () => ({ text }: { text: string }) => <div>{text}</div>);

describe('CardsList Component', () => {
  it('renders without crashing', () => {
    render(<CardsList todosList={[]} />);
    expect(screen.getByText('There are no tasks yet...')).toBeInTheDocument();
  });

  it('renders multiple Card components', () => {
    const todosList = [
      { id: 1, name: 'Task 1' },
      { id: 2, name: 'Task 2' },
    ];

    render(<CardsList todosList={todosList} />);

    expect(screen.getByText('Task 1')).toBeInTheDocument();
    expect(screen.getByText('Task 2')).toBeInTheDocument();
  });

  it('renders default message when todosList is empty', () => {
    render(<CardsList todosList={[]} />);
    expect(screen.getByText('There are no tasks yet...')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const todosList = [
      { id: 1, name: 'Task 1' },
      { id: 2, name: 'Task 2' },
    ];

    const { container } = render(<CardsList todosList={todosList} />);
    expect(container).toMatchSnapshot();
  });
});
