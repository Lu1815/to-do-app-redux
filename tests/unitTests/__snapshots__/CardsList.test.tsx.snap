// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList Component matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col justify-center gap-2 py-3"
  >
    <div>
      Task 1
    </div>
    <div>
      Task 2
    </div>
  </div>
</div>
`;
